File: 696-count-binary-substrings/count-binary-substrings.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        n = len(s)
        pre, cur = 0, 1
        count = 0

        for i in range(1, n):
            if s[i] == s[i - 1]:
                cur += 1
            else:
                pre = cur
                cur = 1

            if pre >= cur:
                count += 1

        return count

File: 696-count-binary-substrings/count-binary-substrings.cs
public class Solution {
    public int CountBinarySubstrings(string s) {
        int n = s.Length;
        int count = 0, prev = 0, curr = 1;
        
        for (int i = 1; i < n; i++) {
            if (s[i] == s[i - 1]) {
                curr++;
            }
            else {
                count += Math.Min(prev, curr);
                prev = curr;
                curr = 1;
            }
        }
        
        count += Math.Min(prev, curr);
        return count;
    }
}